persist-cli/src/main/java/io/ballerina/persist/utils/BalProjectUtils.java

```diff
@@ -56,60 +51,55 @@ public class BalProjectUtils {
 
     private BalProjectUtils() {}
 
-    public static DiagnosticResult hasSemanticDiagnostics(Path projectPath) {
-        Package currentPackage;
-        BuildProject buildProject;
-        buildProject = BuildProject.load(projectPath.toAbsolutePath());
-        currentPackage = buildProject.currentPackage();
-        PackageCompilation compilation = currentPackage.getCompilation();
-        return compilation.diagnosticResult();
-    }
-
     public static EntityMetaData getEntitiesInBalFiles(String sourcePath) throws BalException {
-        ArrayList<Entity> returnMetaData = new ArrayList<>();
-        ArrayList<ModuleMemberDeclarationNode> returnModuleMembers = new ArrayList<>();
+        ArrayList<Entity> entityMetaData = new ArrayList<>();
+        ArrayList<ModuleMemberDeclarationNode> entityModuleMembers = new ArrayList<>();
         Path dirPath = Paths.get(sourcePath);
-        List<Path> fileList;
-        Path clientEntitiesPath = Paths.get(sourcePath, KEYWORD_MODULES, KEYWORD_CLIENTS,
-                PATH_ENTITIES_FILE).toAbsolutePath();
-        File entitiesBal = new File(clientEntitiesPath.toString());
         try {
-            if (!entitiesBal.exists()) {
-                DiagnosticResult diagnosticResult = hasSemanticDiagnostics(dirPath);
-                if (diagnosticResult.hasErrors()) {
-                    StringBuilder errorMessage = new StringBuilder();
-                    errorMessage.append("Error occurred when validating the project. ");
-                    for (Diagnostic d : diagnosticResult.errors()) {
+            BuildProject buildProject = BuildProject.load(dirPath.toAbsolutePath());;
+            Package currentPackage = buildProject.currentPackage();
+            PackageCompilation compilation = currentPackage.getCompilation();
+            DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+            if (diagnosticResult.hasErrors()) {
+                StringBuilder errorMessage = new StringBuilder();
+                errorMessage.append("Error occurred when validating the project. ");
+                int validErrors = 0;
+                for (Diagnostic diagnostic : diagnosticResult.errors()) {
+                    if (!diagnostic.diagnosticInfo().code().equals("BCE2066")) {
```

Comment:
Lets try not to hardcode constants from dependency. There should be Ballerina-lang level constants we can use.

Reply:
Updated in [0cc9a4b](https://github.com/ballerina-platform/persist-tools/pull/70/commits/0cc9a4bc0c186b7b8b3d19e9168a981ed56c4d59)